src/views/Activity.vue

```diff
@@ -0,0 +1,428 @@
+      for (var label in chunks){
+        var slice_i = label.indexOf(':')
+        var labeltype = label.slice(0,slice_i);
+        var labelvalue = label.slice(slice_i+1);
+        if (chunks[label]['duration'] === undefined){
+          console.error("Chunk has no duration: "+Object.keys(chunks[label]));
+        }
+        else if (labeltype == "appname"){
+          applabels.push([labelvalue, label]);
+        }
+        else if (labeltype == "title"){
+          titlelabels.push([labelvalue, label]);
+        }
+      }
+      //console.log(applabels);
+      //console.log(titlelabels);
+
+      // Add apps
+      var apps = {};
+      for (var i in applabels){
+        var name = applabels[i][0];
+        var label = applabels[i][1];
+        apps[name] = {
+          'name':     name,
+          'duration': chunks[label]['duration'],
+          'titles': [],
+        };
+      }
+      //console.log(apps);
+
+      // Add titles
+      for (var i in titlelabels){
+        var name = titlelabels[i][0];
+        var label = titlelabels[i][1];
+        // Find appname from sublabels
+        var appname = null;
+        for (var i in chunks[label]['other_labels']){
+          var sublabel = chunks[label]['other_labels'][i];
+          var slice_i = sublabel.indexOf(':');
+          var sublabeltype = sublabel.slice(0,slice_i);
+          var sublabelvalue = sublabel.slice(slice_i+1);
+          if (sublabeltype == "appname" && sublabelvalue in apps){
+            appname = sublabelvalue;
+          }
+        }
+        if (appname){
+          apps[appname]['titles'].push({
+            'name': name,
+            'duration': chunks[label]['duration'],
+          });
+          //console.log("Added title to "+appname+": "+name);
+        }
+      }
+      //console.log(apps);
+
+      // Sort titles by duration
+      for (var app in apps){
+        apps[app]['titles'].sort(function(a,b){
+          if (a['duration']['value'] < b['duration']['value'])
+            return 1;
+          else
+            return -1;
+        });
+      }
+      // Convert apps to a list and sort by duration
+      var applist = [];
+      for (var app in apps){
+        applist.push(apps[app]);
+      }
+      applist.sort(function(a,b){
+        if (a['duration']['value'] < b['duration']['value'])
+          return 1;
+        else
+          return -1;
+      });
+      // Convert second duration human readable form
+      for (var appi in applist){
+        var app = applist[appi];
+        //console.log(app);
+        app['duration'] = this.secondsToDuration(app['duration']['value']);
+        for (var titlename in app['titles']){
+          var title = app['titles'][titlename];
+          title['duration'] = this.secondsToDuration(title['duration']['value']);
+        }
+      }
+      // Set list
+      this.$set("appsummary", applist);
+    },
+
+
+    /*
+
+        Date/time functions
+
+    */
+
+
+    setDay: function(datestr){
```

Comment:
We have [moment.js](http://momentjs.com/) as a dependency, it's a lot more reasonable to use.